velox/substrait/SubstraitToVeloxPlan.cpp

```diff
@@ -1773,13 +1789,40 @@ core::TypedExprPtr SubstraitVeloxPlanConverter::connectWithAnd(
     }
   }
 
+  std::shared_ptr<const core::ITypedExpr> ifThenFilter = nullptr;
+  if (ifThens.size() > 0) {
+    ifThenFilter = exprConverter_->toVeloxExpr(ifThens[0], inputType);
+    uint32_t idx = 1;
+    while (idx < ifThens.size()) {
+      ifThenFilter = connectWithAnd(
+          ifThenFilter, exprConverter_->toVeloxExpr(ifThens[idx], inputType));
+      idx += 1;
+    }
+  }
+
   VELOX_CHECK(
-      scalarFilter != nullptr || orListFilter != nullptr,
+      scalarFilter != nullptr || orListFilter != nullptr ||
+          ifThenFilter != nullptr,
       "One filter should be valid.");
   if (scalarFilter != nullptr && orListFilter != nullptr) {
```

Comment:
How about we improving this part of logic a bit to make it more readable? A function which returns all the non-null filters can be added, and connectWithAnd can be then used for the final return.